cocotb/handle.py

```diff
@@ -565,14 +563,20 @@ def drivers(self):
         iterator = simulator.iterate(self._handle, simulator.DRIVERS)
         while True:
             # Path is left as the default None since handles are not derived from the hierarchy
-            yield SimHandle(simulator.next(iterator))
+            try:
+                yield SimHandle(simulator.next(iterator))
+            except StopIteration:
+                return
 
     def loads(self):
         """An iterator for gathering all loads on a signal."""
         iterator = simulator.iterate(self._handle, simulator.LOADS)
         while True:
             # Path is left as the default None since handles are not derived from the hierarchy
-            yield SimHandle(simulator.next(iterator))
+            try:
+                yield SimHandle(simulator.next(iterator))
+            except StopIteration:
+                return
```

Comment:
Strictly this would be better as:
```
try:
    v = simulator.next(iterator)
except StopIteration:
    return
yield SimHandle(v)
```
as this doesn't swallow unexpected exceptions from `SimHandle` or `yield`. Same throughout.

A possibly neater option would be
```python
class SimIterator(collections.abc.Iterator):
    def __init__(self, handle, mode):
        self._iterator = simulator.iterate(handle, mode)
    def __next__(self):
        return simulator.next(self._iterator)
    if sys.version_info.major < 3:
        next = __next__
```

making this function simply `return SimIterator(self._handle, simulator.LOADS)`

Reply:
The class is one more thing to document that's only being used in 2 places. But you are definitely right about the first bit.

Reply:
It wouldn't need documenting - it's a private type just like the built-in "dict_iterator" and "list_iterator".

Reply:
Could call it `_SimIterator` to emphasie that